redshift/resource_redshift_user.go

```diff
@@ -292,37 +291,34 @@ func resourceRedshiftUserUpdate(d *schema.ResourceData, meta interface{}) error
 
 	err := readRedshiftUser(d, tx)
 
-	if err == nil {
-		tx.Commit()
-		return nil
-	} else {
+	if err != nil {
 		tx.Rollback()
 		return err
 	}
+
+	tx.Commit()
+	return nil
 }
 
 func resetPassword(tx *sql.Tx, d *schema.ResourceData, username string) error {
 
 	if v, ok := d.GetOk("password_disabled"); ok && v.(bool) {
-
 		var disablePasswordQuery = "alter user " + username + " password disable"
 
 		if _, err := tx.Exec(disablePasswordQuery); err != nil {
 			return err
 		}
 		return nil
+	}
 
-	} else {
```

Comment:
I personally find it easier to follow the logic with the else branch, even if it is redundant

Reply:
I have reverted the change to `resetPassword()`.